vividus/src/main/java/org/vividus/configuration/VaultStoredPropertiesProcessor.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2019-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.configuration;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.Optional;
+import java.util.Properties;
+
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.core.env.PropertiesPropertySource;
+import org.springframework.vault.config.EnvironmentVaultConfiguration;
+import org.springframework.vault.core.VaultKeyValueOperationsSupport.KeyValueBackend;
+import org.springframework.vault.core.VaultTemplate;
+import org.springframework.vault.support.VaultResponse;
+
+public class VaultStoredPropertiesProcessor extends AbstractPropertiesProcessor implements Closeable
+{
+    private final Properties properties;
+    private VaultTemplate vaultTemplate;
+    private AnnotationConfigApplicationContext context;
+
+    VaultStoredPropertiesProcessor(Properties properties)
+    {
+        super("VAULT");
+        this.properties = properties;
+    }
+
+    @Override
+    protected String processValue(String propertyName, String partOfPropertyValueToProcess)
+    {
+        int indexOfEngineSeparator = partOfPropertyValueToProcess.indexOf('/');
+        int indexOfKeySeparator = partOfPropertyValueToProcess.lastIndexOf('/');
+        String engine = partOfPropertyValueToProcess.substring(0, indexOfEngineSeparator);
+        String path = partOfPropertyValueToProcess.substring(indexOfEngineSeparator + 1, indexOfKeySeparator);
+        String key = partOfPropertyValueToProcess.substring(indexOfKeySeparator + 1);
```

Comment:
should not we add validation of partOfPropertyValueToProcess? just to not accidentally get some out of bound index